java/rocksjni/portal.h

```diff
@@ -3668,6 +3669,55 @@ class FilterPolicyJni
     }
     return kUnknownFilterPolicy;
   }
+
+  /**
+   * Create a new Java org.rocksdb.Filter object with the
+   * properties as the provided C++ ROCKSDB_NAMESPACE::FilterPolicy
+   * object
+   *
+   * @param env A pointer to the Java environment
+   * @param filter_policy A pointer to ROCKSDB_NAMESPACE::FilterPolicy object
+   *
+   * @return A reference to a Java org.rocksdb.Filter object, or
+   * nullptr if an an exception occurs
+   */
+  static jobject construct(
+      JNIEnv* env, const std::shared_ptr<const FilterPolicy>& filter_policy) {
+    auto filter_policy_typejni = FilterPolicyJni::getFilterPolicyType(
+        filter_policy->CompatibilityName());
+    jclass jclazz = nullptr;
+    jmethodID method_id_init = nullptr;
+    switch (filter_policy_typejni) {
+      case kBloomFilterPolicy:
+        jclazz = RocksDBNativeClass::getJClass(env, "org/rocksdb/BloomFilter");
+        if (jclazz != nullptr) {
+          method_id_init = env->GetMethodID(jclazz, "<init>", "(JD)V");
+        }
+        break;
+      default:
+        break;
+    }
+
+    if (jclazz == nullptr || method_id_init == nullptr) {
+      // not a policy type implemented in Java
+      // OR exception occurred accessing class / constructor
+      // exception thrown: NoSuchMethodException or OutOfMemoryError
+      return nullptr;
+    }
+
+    auto* sptr_filter =
+        new std::shared_ptr<const ROCKSDB_NAMESPACE::FilterPolicy>(
+            filter_policy);
+
+    double filter_bits = 0.0;
+    if (const ROCKSDB_NAMESPACE::BloomLikeFilterPolicy* bloomlike_filter =
+            dynamic_cast<const BloomLikeFilterPolicy*>(sptr_filter->get())) {
+      filter_bits = bloomlike_filter->GetMillibitsPerKey() / 1000.0;
```

Comment:
I slightly don't like to store `filter_bits` in Java when it's also in C++. But I understand that we already have this in code and probably there are reasons. Maybe `hashCode` & `equals` methods must be fast in Java and should not cross native code boundary to read `filter_bits`.

Reply:
Yes, just want to report/configure the bit number on the Java side, so I think it works easiest for it to be there.